Cabal/doc/nix-local-build.rst

```diff
@@ -419,6 +419,28 @@ have to separate them with ``--``.
 
     $ cabal new-run target -- -a -bcd --argument
 
+'new-run' also supports running script files that use a certain format. With
+a script that looks like:
+
+::
+
+    #!/usr/bin/env cabal
+    {- cabal:
+    build-depends: base ^>= 4.11
+                , shelly ^>= 1.8.1
```

Comment:
Probably needs `default-language: Haskell2010` as well, since otherwise you always get a warning. Unless we decide to splice that in automagically.

Reply:
@23Skidoo actually this makes me think of something... I think we should encode the spec-version somehow... e.g.

```
#!/usr/bin/env cabal
{-# LANGUAGE Haskell2010, GADTs #-}
{- cabal:2.4
build-depends: ...
-}
```

and we might want to tolerate *not* having to specify `default-language` w/ scripts, as `default-{extensions,language}` makes less sense imho for scripts, as we have actual language pragmas (I've intentionally added one in the example above), and you certainly wouldn't want to replicate those...

Reply:
What about multi-file scripts? Easier to set `default-language` in one place than use a pragma in each file.

Reply:
@23Skidoo we can still support `default-language`; I just think we shouldn't force people to use it by emitting nasty warnings like we do for `.cabal` files...

Reply:
It skips that check on purpose.

Reply:
@23Skidoo depends on the semantics you want to have for scripts... 

However, I suggest we postpone this design/bikeshedding issue to a followup-task in order not to hold up merging this PR; I think all @typedrat needs to do is tweaks the docs and we're ready to go?